### PR TITLE
Test cases - Issue #29

### PR DESCRIPTION
**Pull request by moekraft**
Original draft of Test cases section per issue #29.

### CODE REVIEW

**Comment by moekraft** [comment]
Question: How do I nest lists? I tried the recommended Markdown format but list did not nest. See http://commonmark.org/help/tutorial/10-nestedLists.html. Does bikeshed support nested lists?

**Comment by maryjom** [comment]
Added to the [22 Feb survey](https://www.w3.org/2002/09/wbs/93339/ACTTF22Feb2017/).

File: act-framework.bs [hunk 1]
```diff
@@ -131,7 +131,15 @@ Selector syntax depends on the input type. When the input data is an HTML docume
...
 
-Editor note: This section describes how rules are broken down into one or more test cases. Each test case gives some result that, when combined, provides the outcome of the rule. Additionally this section describes how rule authors should write test cases, and the mechanism of combining their outcomes.
+Each rule contains one or more testcases written in plain language. Each testcase MUST provide the following:
+
+* Testcase mode identifying whether testcase steps are automated, semi-automated or require manual testing.
+* Testcase steps which clearly outline the steps necessary to reach the result.
+* Test case result indicating whether testcase passed or failed.
+  * Description of error causing any failing result.
+
+Test case results are combined with other test case results to provide a cumulative outcome, pass or fail, of the rule.
```
Comment (WilcoFiers) [inline act-framework.bs]: I suggest we update this to the following:

"Test case results are combined with other test case results to provide a cumulative outcome, pass or fail, of the rule. When a rule has multiple test cases, the results of all test cases are combined to give a single for each selected item. The results are cumulative, meaning that as long as one test case passes, the rule has passed.

Example:
A rule for HTML img elements has two test cases, one that checks if there is a text alternative, and another that checks if the image is marked as decorative. If either one of these passed, the rule has passed. Only if both fail, does that element fail the rule."

File: act-framework.bs [hunk 2]
```diff
@@ -131,7 +131,15 @@ Selector syntax depends on the input type. When the input data is an HTML docume
 Test Cases {#test-proc-cases}
 ---------------------------------------
 
-Editor note: This section describes how rules are broken down into one or more test cases. Each test case gives some result that, when combined, provides the outcome of the rule. Additionally this section describes how rule authors should write test cases, and the mechanism of combining their outcomes.
+Each rule contains one or more testcases written in plain language. Each testcase MUST provide the following:
+
+* Testcase mode identifying whether testcase steps are automated, semi-automated or require manual testing.
+* Testcase steps which clearly outline the steps necessary to reach the result.
+* Test case result indicating whether testcase passed or failed.
+  * Description of error causing any failing result.
```
Comment (WilcoFiers) [inline act-framework.bs]: Should be "of the error".

File: act-framework.bs [hunk 3]
```diff
@@ -131,7 +131,18 @@ Selector syntax depends on the input type. When the input data is an HTML docume
 Test Cases {#test-proc-cases}
 ---------------------------------------
 
-Editor note: This section describes how rules are broken down into one or more test cases. Each test case gives some result that, when combined, provides the outcome of the rule. Additionally this section describes how rule authors should write test cases, and the mechanism of combining their outcomes.
+Each rule contains one or more testcases written in plain language. Each testcase MUST provide the following:
```
Comment (moekraft) [inline act-framework.bs]: From today's meeting we are going with test cases. I will check the document, update and merge the PR. Thanks, Moe

File: act-framework.bs [hunk 4]
```diff
@@ -131,7 +131,18 @@ Selector syntax depends on the input type. When the input data is an HTML docume
...
+
+* Testcase mode identifying whether testcase steps are automated, semi-automated or require manual testing.
+* Testcase steps which clearly outline the steps necessary to reach the result.
+* Test case result indicating whether testcase passed or failed.
+  * Description of the error causing any failing result.
+
```
Comment (moekraft) [inline act-framework.bs]: From today's meeting, the decision was that did not make sense under test cases but is covered elsewhere.

**Pull request merged**